ts/components/IOMarkdown/markdownRenderer.ts

```diff
@@ -63,3 +80,129 @@ function integrateParent<T extends AnyTxtNode>(
       }
     : { ...node, parent: parentLight };
 }
+
+export const sanitizeMarkdownForImages = (
+  inputMarkdownContent: string
+): string => {
+  const markdownImageRegex = /!\[.*?\]\((.*?)\)/g;
+
+  const reversedMatches: Array<RegExpExecArray> = [];
+  // eslint-disable-next-line functional/no-let
+  let match: RegExpExecArray | null;
+  while ((match = markdownImageRegex.exec(inputMarkdownContent)) !== null) {
+    // eslint-disable-next-line functional/immutable-data
+    reversedMatches.push(match);
+  }
+  // eslint-disable-next-line functional/immutable-data
+  reversedMatches.reverse();
```

Comment:
```suggestion
    // eslint-disable-next-line functional/immutable-data
    reversedMatches.unshift(match);
  }
```
How about using `unshift` to avoid the need for `reverse` at the end of the iteration? What do you think?